pages/account/statistics.js

```diff
@@ -66,6 +67,40 @@ export async function getServerSideProps(context) {
     logger.error(e, "ERROR get user's account statistics");
   }
 
+  const dailyStatsMap = new Map();
+  const currentDate = new Date();
+  const last30Days = [];
+
+  const totalClicks = data.links.individual.reduce((acc, link) => {
+    return acc + link.clicks;
+  }, 0);
+  data.links.clicks = totalClicks;
+
+  // Loop through daily stats data and populate dailyStatsMap
+  data.profile.daily.forEach(stat => {
+    dailyStatsMap.set(
+      dateFormat({ format: "short", date: new Date(stat.date) }),
+      stat.value
+    );
+  });
+
+  // Populate last30Days array
+  for (let index = 0; index < 30; index++) {
+    const date = new Date(currentDate);
+    date.setDate(currentDate.getDate() - index);
+    const formattedDate = dateFormat({ format: "short", date: date });
+    last30Days.unshift(formattedDate);
+
+    if (!dailyStatsMap.has(formattedDate)) {
+      dailyStatsMap.set(formattedDate, { views: 0 });
+    }
+  }
+
+  const last30DaysStats = last30Days.map(date => ({
+    views: dailyStatsMap.has(date) ? dailyStatsMap.get(date).views : 0,
```

Comment:
If a date is missing, I get the following error...

<img width="1060" alt="Screenshot 2023-08-26 at 08 44 53" src="https://github.com/EddieHubCommunity/LinkFree/assets/624760/abe07786-d5e2-4f3d-b997-f0bc06972026">

Reply:
I have made the necessary changes, please check.